### ISSUE TITLE
ExpressionChangedAfter ... on modal open from inside a form.

### ISSUE THREAD

**Issue by giniedp** [ng-bootstrap/ng-bootstrap#1972]
### Bug description:

ExpressionChangedAfterItHasBeenCheckedError is raised, when a modal content is opened via "submit" event ("enter" key) from inside a form, when the text input is focused.

I tried to trace down the responsible property and i think it is the "ng-focused" class of the input, that is going crazy when the enter key is pressed. But i am not sure.

### Link to minimally-working plunker that reproduces the issue:

http://plnkr.co/edit/hfnFJUjKXIyJvsLcUXWe?p=info

focus the input field and hit the "enter" key. The error is logged only the first time. To reproduce, reload the plnkr and repeat.

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 5.0.2

ng-bootstrap: 1.0.0-beta.1

Bootstrap: 4.0.0-beta.2

**Comment by jeffreypolk** [comment]
Not sure if it's related, but I get the same error when I try to use an accordion within a modal.  The accordion is not static, it is built with *ngFor and interpolation.

**Comment by pkozlowski-opensource** [comment]
This is duplicate of #1252. I'm afraid that we don't have any great solution here - your best option for now is to wrap modal opening logic into `setTimeout(...)`...

**Comment by giniedp** [comment]
`setTimeout` on my site does not help. As i understand the issue #1252 its the focus of the modal window that causes error. So it does not matter whether i delay my call to modal.open or not. As soon as the modal window switches the focus from my input to the window, the error occurs.

I think the `setTimeout` fix should be applied inside the bootstrap library. Looking at https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/modal/modal-window.ts#L65
it must be changed from this

```ts
  ngAfterViewInit() {
    if (!this._elRef.nativeElement.contains(document.activeElement)) {
      this._elRef.nativeElement['focus'].apply(this._elRef.nativeElement, []);
    }
  }
```

to this (i use `Promise.resolve().then` instead of `setTimeout`)

```ts
  ngAfterViewInit() {
    if (!this._elRef.nativeElement.contains(document.activeElement)) {
      Promise.resolve().then(() => {
        this._elRef.nativeElement['focus'].apply(this._elRef.nativeElement, []);
      })
    }
  }
```

**Comment by priteshacharya** [comment]
@giniedp your solution worked for me. I wonder when it will go to PR and release.

**Comment by giniedp** [comment]
not sure if this issue ist still tracked by ng-bootstrap, since this one is closed as it is a dublicate of #1252 but #1252 is locked but without a solution.

@pkozlowski-opensource would you mind to review my proposal above?
